dbt_platform_helper/commands/environment.py

```diff
@@ -594,50 +562,54 @@ def add_maintenance_page(
     env: str,
     services: List[Service],
     allowed_ips: tuple,
-    ip_filter: bool,
     template: str = "default",
 ):
     lb_client = session.client("elbv2")
     maintenance_page_content = get_maintenance_page_template(template)
+    bypass_value = "".join(random.choices(string.ascii_lowercase + string.digits, k=12))
+
+    counter = 1
 
     for svc in services:
-        target_group_arn = find_target_group(app, env, svc.name)
+        target_group_arn = find_target_group(app, env, svc.name, session)
 
         # not all of an application's services are guaranteed to have been deployed to an environment
         if not target_group_arn:
             continue
 
-        if not ip_filter:
-            user_ip = get_public_ip()
-            allowed_ips = list(allowed_ips) + [user_ip]
-
+        allowed_ips = list(allowed_ips)
+        for ip_index, ip in enumerate(allowed_ips):
+            forwarded_rule_priority = counter * (ip_index + 100)
```

Comment:
I'm struggling to understand the reasoning behind this maths.

Can you write the criteria for `forwarded_rule_priority` in human facing terms?

Reply:
Each service needs a Bypass-Key rule created. This is assigned the counter priority. Then for each allowed ip for each service, we create an X-Forwarded-For rule.

So say we have two services and three allowed ips.

1st iteration:
- Create bypass rule with priority 1
- Then rules for each ip with priorities: 100, 101, 102

2nd:
- Create bypass rule with priority 2
- Then rules for each ip with priorities: 200, 202, 204

3rd:
- Create bypass rule with priority 3
- Then rules for each ip with priorities: 300, 303, 306

It's just to ensure that there's no chance of clashing priority numbers, even if we have loads of allowed ips

Reply:
Thanks. That confirms the feeling I had on Friday.

I suggest we rename `counter` to `service_number` so it's more obvious what it represents.

Also, 100 is a magic number, please can we extract that to a variable `max_allowed_ips`.

Also, I'm not sure how many allowed IPs we might have, but is 100 per service definitely enough? @lgarvey would have a good feel for that I think.

As things stand, if the second service has >=50 allowed IPs, it will start clashing with the third service...

```
2 * (50 + 100) = 300
```

This will only get worse as the number of services increases.

Perhaps...

```suggestion
            forwarded_rule_priority = (service_number * max_allowed_ips) + ip_index
```

Then...
```
(2 * 100) + 50 = 250
```

Reply:
`max_allowed_ips` could probably just be 1000 and I think we would be covered whilst still having the service number as the first digit.

Reply:
I think 100 is fine. The only ips we're allowing are the env ips and there won't be more than 100 of those